homeassistant/components/media_player/dunehd.py

```diff
@@ -0,0 +1,171 @@
+            state = STATE_PLAYING
+        if int(self._state.get('playback_speed', 1234)) == 0:
+            state = STATE_PAUSED
+        if self._state['player_state'] == 'navigator':
+            state = STATE_ON
+        return state
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def volume_level(self):
+        """Volume level of the media player (0..1)."""
+        return int(self._state.get('playback_volume', 0)) / 100
+
+    @property
+    def is_volume_muted(self):
+        """Boolean if volume is currently muted."""
+        return int(self._state.get('playback_mute', 0)) == 1
+
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        return list(self._sources.keys())
+
+    @property
+    def supported_media_commands(self):
+        """Flag of media commands that are supported."""
+        return DUNEHD_PLAYER_SUPPORT
+
+    def volume_up(self):
+        """Volume up media player."""
+        self._state = self._player.volume_up()
+
+    def volume_down(self):
+        """Volume down media player."""
+        self._state = self._player.volume_down()
+
+    def mute_volume(self, mute):
+        """Mute/unmute player volume."""
+        self._state = self._player.mute(mute)
+
+    def turn_off(self):
+        """Turn off media player."""
+        self._media_title = None
+        self._state = self._player.turn_off()
+        self.schedule_update_ha_state()
+
+    def turn_on(self):
+        """Turn off media player."""
+        self._state = self._player.turn_on()
+        self.schedule_update_ha_state()
+
+    def media_play(self):
+        """Play media media player."""
+        self._state = self._player.play()
+        self.schedule_update_ha_state()
+
+    def media_pause(self):
+        """Pause media player."""
+        self._state = self._player.pause()
+        self.schedule_update_ha_state()
+
+    @property
+    def media_title(self):
+        """Current media source."""
+        self.__update_title()
+        if self._media_title:
+            return self._media_title
+        return self._state.get('playback_url', 'Not playing')
+
+    def __update_title(self):
+        if self._state['player_state'] == 'bluray_playback':
+            self._media_title = 'Blu-Ray'
+        elif 'playback_url' in self._state:
+            sources = self._sources
+            sval = sources.values()
+            skey = sources.keys()
+            pburl = self._state['playback_url']
+            if pburl in sval:
+                self._media_title = list(skey)[list(sval).index(pburl)]
+            else:
+                self._media_title = pburl
+
+    def select_source(self, source):
+        """Select input source."""
+        self._media_title = source
+        self._state = self._player.launch_media_url(self._sources.get(source))
+        self.schedule_update_ha_state()
```

Comment:
No need for any of the `schedule_update_ha_state` calls. Since your device does not overwrite `should_poll`, it defaults to `True` which will trigger Home Assistant to immediately call `update()` after a call to any of the services.